editor/import/resource_importer_scene.cpp

```diff
@@ -631,8 +631,9 @@ Node *ResourceImporterScene::_post_fix_node(Node *p_node, Node *p_root, Map<Ref<
 								RigidBody3D *rigid_body = memnew(RigidBody3D);
 								rigid_body->set_name(p_node->get_name());
 								p_node->replace_by(rigid_body);
-								rigid_body->set_transform(mi->get_transform());
+								p_node->queue_delete();
```

Comment:
What's the point to do `p_node->replace_by(rigid_body)` to then free `p_node`, and then assign it `rigid_body`? That's pretty confusing.

Reply:
In the case of p_node, this will be the root of the object which will by default be a Node3D type. For rigid bodies, we first want to use the replace_by function which will replace the Node3D node with a new RigidBody node. The root of this object now be a rigid body, and according to replace_by, the original node still exists, it's simply been removed from the tree.

The reason we call queue_delete on p_node now is since we have removed the original root node, I believe leaving it will cause a memory leak. This was simply to make it consistent with the behaviour of the other two switch cases which do similar things. We then assign p_node the new rigid body node, which has become the new root. This is because lower in the function, p_node is still consider as part of the function.

One thing I could do though is move the rigid_body->set_transform function above rigid_body->set_name which will make it more consistent with the other two switch cases, and hopefully a little less confusing.

Reply:
Hm, I thought that `replace_by` would modify `p_node`'s `this`, which is why I was confused that we'd free the modified instance. But indeed it doesn't change `p_node`, it takes its place and sets groups and properties accordingly, but leave `p_node` unchanged (and orphaned).

It's a bit of a weird API IMO, but the importer code makes sense then. Thanks for the explanations :)